crml/cennzx/src/benchmarking.rs

```diff
@@ -0,0 +1,160 @@
+/* Copyright 2019-2020 Centrality Investments Limited
+*
+* Licensed under the LGPL, Version 3.0 (the "License");
+* you may not use this file except in compliance with the License.
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+* You may obtain a copy of the License at the root of this project source code,
+* or at:
+*     https://centrality.ai/licenses/gplv3.txt
+*     https://centrality.ai/licenses/lgplv3.txt
+*/
+
+//! Attestation benchmarking.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::*;
+
+use frame_benchmarking::{account, benchmarks, whitelisted_caller};
+use frame_system::RawOrigin;
+
+use crate::Module as Cennzx;
+
+benchmarks! {
+	_{ }
+
+	buy_asset {
+		let investor: T::AccountId = whitelisted_caller();
+		let buyer: T::AccountId = account("buyer", 0, 0);
+
+		let core_asset_id = <Cennzx<T>>::core_asset_id();
+		let asset_a: T::AssetId = 2u32.into();
+		let asset_b: T::AssetId = 3u32.into();
+
+		let _ = T::MultiCurrency::deposit_creating(&investor, Some(core_asset_id), 1000u32.into());
+		let _ = T::MultiCurrency::deposit_creating(&investor, Some(asset_a), 200u32.into());
+		let _ = T::MultiCurrency::deposit_creating(&investor, Some(asset_b), 300u32.into());
+		let _ = T::MultiCurrency::deposit_creating(&buyer, Some(asset_a), 100u32.into());
+
+		let _ = <Cennzx<T>>::add_liquidity(RawOrigin::Signed(investor.clone()).into(), asset_a, 20u32.into(), 20u32.into(), 100u32.into());
+		let _ = <Cennzx<T>>::add_liquidity(RawOrigin::Signed(investor.clone()).into(), asset_b, 30u32.into(), 30u32.into(), 100u32.into());
+
+	}: _(RawOrigin::Signed(buyer.clone()), None, asset_a, asset_b, 10u32.into(), 50u32.into())
+	verify {
+		assert_eq!(T::MultiCurrency::free_balance(&buyer, Some(asset_a)), 79u32.into());
+	}
+
+	sell_asset {
+		let investor: T::AccountId = whitelisted_caller();
+		let seller: T::AccountId = account("seller", 0, 0);
+
+		let core_asset_id = <Cennzx<T>>::core_asset_id();
+		let asset_a: T::AssetId = 2u32.into();
+		let asset_b: T::AssetId = 3u32.into();
+
+		let _ = T::MultiCurrency::deposit_creating(&investor, Some(core_asset_id), 1000u32.into());
```

Comment:
wondering if using some larger values would sway the benchmarks in any meaningful way?
these numbers are a very small amount compared to real use
especially add/remove liquidity could be > 1,000,000.0000 easily

Reply:
The numbers, though small, are converted to and held in large number containers. Thus my understanding is that shouldn't make a difference.